Assets/General/Scripts/Story/StoryManager.cs

```diff
@@ -0,0 +1,176 @@
+            }
+        }
+    }
+
+    /// <summary>
+    /// Start a specific story beat's events over time
+    /// </summary>
+    /// <param name="events">The events to run</param>
+    /// <returns></returns>
+    private IEnumerator TriggerBeatEvents(List<StoryBeatEvent> events)
+    {
+        WaitForSeconds postEventDelay;
+
+        // Iterate through each event
+        for (int i = 0; i < events.Count; i++)
+        {
+            // Trigger the event, then start a delay before triggering the next event
+            events[i].TriggerEvent();
+            postEventDelay = new(events[i].DelayTime);
+            yield return postEventDelay;
+        }
+    }
+}
+
+/// <summary>
+/// An individual story beat
+/// </summary>
+[System.Serializable]
+public class StoryBeat
+{
+    // The name of the story beat
+    public string Name;
+
+    // The list of story beat events in the beat
+    public List<StoryBeatEvent> StoryBeatEvents;
+}
+
+/// <summary>
+/// An individual story beat event
+/// </summary>
+[System.Serializable]
+public class StoryBeatEvent
+{
+    // Enumerable for setting the type of beat event
+    public enum BeatEventType { Dialogue, BoatSpeed, BossAttack, Function }
+
+    // Editor settings
+    public bool Minimized;
+
+    // General settings
+    public string Name;
+    public BeatEventType EventType;
+    public float DelayTime;
+
+    // Dialogue settings
+    public ScriptableDialogueUI Dialogue;
+
+    // Boat speed settings
+    public float BoatSpeed;
+    public float SpeedChangeTime;
+
+    // Boss attack settings
+    public List<BaseBossAttack> BossAttacks;
+
+    // Function settings
+    public UnityEvent BeatEvent;
+
+    /// <summary>
+    /// Start the story beat event
+    /// </summary>
+    public void TriggerEvent()
+    {
+        // Based on the event type, do a specific action
+        switch (EventType)
+        {
+            // If it's a dialogue event, start the dialogue
+            case BeatEventType.Dialogue:
+                GameStateManager.Instance.GetOnNewDialogueChain()?.Invoke(Dialogue);
+                break;
+
+            // If it's a boat speed event, change the speed of the boat
+            case BeatEventType.BoatSpeed:
+                Object.FindObjectOfType<BoatMover>().ChangeSpeed(BoatSpeed, SpeedChangeTime);
+                break;
+
+            // If it's a boss attack event, start the attacks of all the individual attackers
+            case BeatEventType.BossAttack:
+                foreach (BaseBossAttack currentAttack in BossAttacks)
+                {
+                    currentAttack.InvokeAttackBegin();
+                }
+                break;
+
+            // If it's a UnityEvent event, invoke the UnityEvent
+            case BeatEventType.Function:
+                BeatEvent.Invoke();
+                break;
+
+            // If it's not an event that can exist, print a little smiley face
+            default:
+                Debug.Log(":P");
```

Comment:
I know this is literally impossible to reach, but regardless this should probably be a Debug.LogError and could have a more descriptive thing to print than a smiley face. For example if someone added a new BeatEventType, but forgot to update this switch statement this could occur